build-system/pr-check/build.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script builds the AMP runtime.
+ * This is run during the CI stage = build; job = build.
+ */
+
+const colors = require('ansi-colors');
+const {
+  printChangeSummary,
+  startTimer,
+  stopTimer,
+  timedExecOrDie: timedExecOrDieBase,
+  uploadBuildOutput} = require('./utils');
+const {determineBuildTargets} = require('./build-target');
+const {getStderr} = require('../exec');
+const {gitDiffColor} = require('../git');
+const {isTravisPullRequestBuild} = require('../travis');
+const FILENAME = 'build.js';
+const FILELOGPREFIX = colors.bold(colors.yellow(`${FILENAME}:`));
+const timedExecOrDie =
+  (cmd, unusedFileName) => timedExecOrDieBase(cmd, FILENAME);
+
+/**
+ * Makes sure package.json and yarn.lock are in sync.
+ * @private
+ */
+function runYarnIntegrityCheck_() {
+  const yarnIntegrityCheck = getStderr('yarn check --integrity').trim();
```

Comment:
TODO: After this is merged, create a dummy PR that contains a bad yarn state (edit `package.json` but not `yarn.lock`) and make sure this check fails the build.